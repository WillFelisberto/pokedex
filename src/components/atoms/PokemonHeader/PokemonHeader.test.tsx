import { render, screen } from '@/tests/test-utils';

import { PokemonHeader } from '.';

describe('<PokemonHeader />', () => {
  it('should render the Pokemon sprite', () => {
    const { container } = render(
      <PokemonHeader name="Charizard" spriteUrl="/sprites/charizard.png" types={['fire']} />
    );
    const sprite = screen.getByAltText('Charizard');
    expect(sprite).toBeInTheDocument();
    expect(sprite).toHaveAttribute('src', expect.stringContaining('/sprites/charizard.png'));

    expect(container).toMatchSnapshot();
  });

  it('should render the type icon', () => {
    render(<PokemonHeader name="Pikachu" spriteUrl="/sprites/pikachu.png" types={['electric']} />);
    const typeIcon = screen.getByAltText('electric');
    expect(typeIcon).toBeInTheDocument();
  });

  it('should apply background gradient', () => {
    render(<PokemonHeader name="Bulbasaur" spriteUrl="/sprites/bulbasaur.png" types={['grass']} />);
    const header = screen.getByTestId('pokemon-header');
    expect(header).toHaveStyle('background: linear-gradient');
  });
});
