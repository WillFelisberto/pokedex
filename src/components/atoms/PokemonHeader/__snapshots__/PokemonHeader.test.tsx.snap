// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonHeader /> should render the Pokemon sprite 1`] = `
<div>
  <div
    class="relative w-full h-[307px] flex items-center justify-center rounded-b-3xl overflow-hidden"
    data-testid="pokemon-header"
    style="clip-path: ellipse(60% 65% at 50% 30%);"
  >
    <div
      class="absolute inset-0 flex items-center justify-center  "
    >
      <div
        class="absolute inset-0 w-1/2 bg-gradient-to-l from-white/15 to-transparent"
      />
      <div
        class="absolute top-[10%] "
      >
        <img
          alt="fire"
          height="140"
          src="/icons/png/filled/fire.png"
          style="mask-image: linear-gradient(135deg, rgba(0 0 0 / 50%), rgba(0,0,0,0)); filter: brightness(10);"
          width="140"
        />
      </div>
    </div>
  </div>
  <img
    alt="Charizard"
    class="absolute z-10 left-1/2 transform bottom-0 -translate-x-1/2"
    height="215"
    src="/sprites/charizard.png"
    width="215"
  />
</div>
`;
